server/ec2spotmanager/models.py

```diff
@@ -34,71 +30,60 @@ def get_available_name(self, name, max_length=None):
 
 
 class PoolConfiguration(models.Model):
-    parent = models.ForeignKey("self", blank=True, null=True)
+    parent = models.ForeignKey('self', blank=True, null=True)
     name = models.CharField(max_length=255, blank=False)
     size = models.IntegerField(default=1, blank=True, null=True)
     cycle_interval = models.IntegerField(default=86400, blank=True, null=True)
+    max_price = models.DecimalField(max_digits=12, decimal_places=6, blank=True, null=True)
+    allowed_regions = models.CharField(max_length=1023, blank=True, null=True)
+    userdata_file = models.FileField(storage=OverwritingStorage(location=getattr(settings, 'USERDATA_STORAGE', None)),
+                                     upload_to=get_storage_path, blank=True, null=True)
+    userdata_macros = models.CharField(max_length=4095, blank=True, null=True)
     ec2_key_name = models.CharField(max_length=255, blank=True, null=True)
+    ec2_tags = models.CharField(max_length=1023, blank=True, null=True)
     ec2_security_groups = models.CharField(max_length=255, blank=True, null=True)
-    ec2_instance_types = models.CharField(max_length=4095, blank=True, null=True)
+    ec2_instance_types = models.CharField(max_length=1023, blank=True, null=True)
     ec2_image_name = models.CharField(max_length=255, blank=True, null=True)
-    ec2_userdata_file = models.FileField(storage=OverwritingStorage(
-        location=getattr(settings, 'USERDATA_STORAGE', None)), upload_to=get_storage_path, blank=True, null=True)
-    ec2_userdata_macros = models.CharField(max_length=4095, blank=True, null=True)
-    ec2_allowed_regions = models.CharField(max_length=1023, blank=True, null=True)
-    ec2_max_price = models.DecimalField(max_digits=12, decimal_places=6, blank=True, null=True)
-    ec2_tags = models.CharField(max_length=1023, blank=True, null=True)
     ec2_raw_config = models.CharField(max_length=4095, blank=True, null=True)
+    provider = models.CharField(max_length=255, blank=True, null=True)
```

Comment:
Provider should be a property of the `Instance`, not the `PoolConfiguration`. We can change that later once we actually have multiple providers. In any case, you're missing the migration for this.